src/Symfony/Component/Routing/Router.php

```diff
@@ -227,6 +227,18 @@ public function setConfigCacheFactory(ConfigCacheFactoryInterface $configCacheFa
     }
 
     /**
+     * Returns whether a route with the given name exists.
+     *
+     * @param string $name The route name
+     *
+     * @return bool
+     */
+    public function hasRoute($name)
+    {
+        return $this->getGenerator()->hasRoute($name);
```

Comment:
Since the interface doesn't have this method anymore, you shouldn't directly use it.
`getGenerator()` may return different implementation.